### ISSUE TITLE
clickhouse-keeper Failed to wait RAFT initialization

### ISSUE THREAD

**Issue by wbxsGithub** [ClickHouse/ClickHouse#35479]
> Make sure to check documentation https://clickhouse.yandex/docs/en/ first. If the question is concise and probably has a short answer, asking it in Telegram chat https://telegram.me/clickhouse_en is probably the fastest way to find the answer. For more complicated questions, consider asking them on StackOverflow with "clickhouse" tag https://stackoverflow.com/questions/tagged/clickhouse 

> If you still prefer GitHub issues, remove all this text and ask your question here.

1.Config: 6shard,every shard has two replicas : 
```
node-01  node-02     -->shard1
node-03 node-04     -->shard2
node-05 node-06     -->shard3
```

2.keeper config:
```xml
<yandex>
    <keeper_server>
        <tcp_port>2182</tcp_port>
        <server_id>3</server_id>
        <log_storage_path>/mnt/storage13/keeper_log/keeper</log_storage_path>
        <snapshot_storage_path>/mnt/storage13/keeper_data/keeper/</snapshot_storage_path>
        <raft_configuration>
            <server>
                <id>1</id>
                <hostname>node-02</hostname>
                <port>9444</port>
            </server>
            <server>
                <id>2</id>
                <hostname>node-04</hostname>
                <port>9444</port>
            </server>
            <server>
                <id>3</id>
                <hostname>node-06</hostname>
                <port>9444</port>
            </server>
        </raft_configuration>
        <coordination_settings>
            <raft_logs_level>debug</raft_logs_level>
            <operation_timeout_ms>10000</operation_timeout_ms>
            <session_timeout_ms>30000</session_timeout_ms>
            <force_sync>false</force_sync>
            <startup_timeout>60000</startup_timeout>
            <!-- we want all logs for complex problems investigation -->
            <reserved_log_items>500000</reserved_log_items>
        </coordination_settings>
    </keeper_server>
</yandex>
```
node-02,node-04,node-06 : the "keeper_port.xml" are exist on the three nodes.

3.I started the replicas which names are node-02,node-04.But when I started the node-06,the following error occurred:

```
2022.03.21 22:54:14.463282 [ 150503 ] {} <Information> RaftInstance: peer 1: DC ID 0, node-02:9444, voting member, 1
peer 2: DC ID 0, node-04:9444, voting member, 1
peer 3: DC ID 0, node-06:9444, voting member, 1
my id: 3, voting_member
num peers: 2
2022.03.21 22:54:14.463299 [ 150503 ] {} <Information> RaftInstance: global manager does not exist. will use local thread for commit and append
2022.03.21 22:54:14.463395 [ 150503 ] {} <Information> RaftInstance: wait for HB, for 50 + [1000, 2000] ms
2022.03.21 22:54:14.463492 [ 150555 ] {} <Information> RaftInstance: bg append_entries thread initiated
2022.03.21 22:54:16.203567 [ 150513 ] {} <Warning> RaftInstance: Election timeout, initiate leader election
2022.03.21 22:54:16.203619 [ 150513 ] {} <Information> RaftInstance: [PRIORITY] decay, target 1 -> 1, mine 1
2022.03.21 22:54:16.203643 [ 150513 ] {} <Information> RaftInstance: [ELECTION TIMEOUT] current role: follower, log last term 0, state term 0, target p 1, my p 1, hb dead, pre-vote NOT done
2022.03.21 22:54:16.203665 [ 150513 ] {} <Information> RaftInstance: [PRE-VOTE INIT] my id 3, my role follower, term 0, log idx 0, log term 0, priority (target 1 / mine 1)
2022.03.21 22:54:16.204417 [ 150516 ] {} <Information> RaftInstance: 0x7fa6676c6118 connected to node-04:9444 (as a client)
2022.03.21 22:54:16.204675 [ 150519 ] {} <Information> RaftInstance: 0x7fa6676c5718 connected to node-02:9444 (as a client)
2022.03.21 22:54:16.204774 [ 150523 ] {} <Information> RaftInstance: [PRE-VOTE RESP] peer 2 (X), term 0, resp term 0, my role follower, dead 1, live 0, num voting members 3, quorum 2
2022.03.21 22:54:16.205026 [ 150524 ] {} <Information> RaftInstance: [PRE-VOTE RESP] peer 1 (X), term 0, resp term 0, my role follower, dead 1, live 0, num voting members 3, quorum 2
2022.03.21 22:54:16.205054 [ 150524 ] {} <Error> RaftInstance: [PRE-VOTE DONE] this node has been removed, stepping down
2022.03.21 22:54:17.945803 [ 150526 ] {} <Information> RaftInstance: stepping down (cycles left: 1), skip this election timeout event
2022.03.21 22:54:19.565909 [ 150527 ] {} <Information> RaftInstance: no hearing further news from leader, remove this server from cluster and step down
2022.03.21 22:55:14.520085 [ 150503 ] {} <Error> void DB::KeeperStorageDispatcher::initialize(const Poco::Util::AbstractConfiguration&, bool): Code: 568, e.displayText() = DB::Exception: Failed to wait RAFT initialization, Stack trace (when copying this message, always include the lines below):

0. /data/wangbaixin/ck/21.8.14.5-lts/ClickHouse/build/../contrib/poco/Foundation/src/Exception.cpp:27: Poco::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int) @ 0xe80022d in /mnt/storage13/ck_install/keeper/clickhouse
1. /data/wangbaixin/ck/21.8.14.5-lts/ClickHouse/build/../src/Common/Exception.cpp:58: DB::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int, bool) @ 0x44061ba in /mnt/storage13/ck_install/keeper/clickhouse
2. /data/wangbaixin/ck/21.8.14.5-lts/ClickHouse/build/../contrib/libcxx/include/string:2218: DB::KeeperServer::waitInit() (.cold) @ 0x3ef455c in /mnt/storage13/ck_install/keeper/clickhouse
3. /data/wangbaixin/ck/21.8.14.5-lts/ClickHouse/build/../src/Coordination/KeeperStorageDispatcher.cpp:262: DB::KeeperStorageDispatcher::initialize(Poco::Util::AbstractConfiguration const&, bool) @ 0xb9b6252 in /mnt/storage13/ck_install/keeper/clickhouse
4. /data/wangbaixin/ck/21.8.14.5-lts/ClickHouse/build/../src/Interpreters/Context.cpp:1720: DB::Context::initializeKeeperStorageDispatcher() const @ 0xaedcc06 in /mnt/storage13/ck_install/keeper/clickhouse
5. /data/wangbaixin/ck/21.8.14.5-lts/ClickHouse/build/../contrib/libcxx/include/vector:1533: DB::Server::main(std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) @ 0x435d29e in /mnt/storage13/ck_install/keeper/clickhouse
6. /data/wangbaixin/ck/21.8.14.5-lts/ClickHouse/build/../contrib/poco/Util/src/Application.cpp:334: Poco::Util::Application::run() @ 0xe769a05 in /mnt/storage13/ck_install/keeper/clickhouse
7. /data/wangbaixin/ck/21.8.14.5-lts/ClickHouse/build/../programs/server/Server.cpp:171: mainEntryClickHouseServer(int, char**) @ 0x4475592 in /mnt/storage13/ck_install/keeper/clickhouse
8. /data/wangbaixin/ck/21.8.14.5-lts/ClickHouse/build/../contrib/libcxx/include/vector:463: main @ 0x4351f60 in /mnt/storage13/ck_install/keeper/clickhouse
9. __libc_start_main @ 0x202e1 in /lib/x86_64-linux-gnu/libc-2.24.so
10. _start @ 0x440385a in /mnt/storage13/ck_install/keeper/clickhouse
 (version 21.8.14.5)

2022.03.21 22:55:14.522260 [ 150503 ] {} <Information> RaftInstance: shutting down raft core
2022.03.21 22:55:14.522289 [ 150503 ] {} <Information> RaftInstance: sent stop signal to the commit thread.
2022.03.21 22:55:14.522298 [ 150503 ] {} <Information> RaftInstance: cancelled all schedulers.
2022.03.21 22:55:14.522310 [ 150503 ] {} <Information> RaftInstance: commit thread stopped.
2022.03.21 22:55:14.522319 [ 150503 ] {} <Information> RaftInstance: all pending commit elements dropped.
2022.03.21 22:55:14.522327 [ 150503 ] {} <Information> RaftInstance: reset all pointers.
2022.03.21 22:55:14.522375 [ 150503 ] {} <Information> RaftInstance: joined terminated commit thread.
2022.03.21 22:55:14.522394 [ 150503 ] {} <Information> RaftInstance: sent stop signal to background append thread.
2022.03.21 22:55:14.522400 [ 150555 ] {} <Information> RaftInstance: bg append_entries thread terminated
2022.03.21 22:55:14.522536 [ 150503 ] {} <Information> RaftInstance: clean up auto-forwarding queue: 0 elems
2022.03.21 22:55:14.522548 [ 150503 ] {} <Information> RaftInstance: clean up auto-forwarding clients
2022.03.21 22:55:14.522556 [ 150503 ] {} <Information> RaftInstance: raft_server shutdown completed.
2022.03.21 22:55:14.522633 [ 150529 ] {} <Error> RaftInstance: failed to accept a rpc connection due to error 125
2022.03.21 22:55:14.522741 [ 150513 ] {} <Information> RaftInstance: end of asio worker thread, remaining threads: 15

2022.03.21 22:55:14.522741 [ 150513 ] {} <Information> RaftInstance: end of asio worker thread, remaining threads: 15
2022.03.21 22:55:14.522688 [ 150532 ] {} <Information> RaftInstance: end of asio worker thread, remaining threads: 38
2022.03.21 22:55:14.522803 [ 150524 ] {} <Information> RaftInstance: end of asio worker thread, remaining threads: 3
2022.03.21 22:55:14.522803 [ 150523 ] {} <Information> RaftInstance: end of asio worker thread, remaining threads: 5
2022.03.21 22:55:14.522807 [ 150527 ] {} <Information> RaftInstance: end of asio worker thread, remaining threads: 1
2022.03.21 22:55:14.522806 [ 150528 ] {} <Information> RaftInstance: end of asio worker thread, remaining threads: 36
...
...
2022.03.21 22:55:14.533395 [ 150503 ] {} <Error> Application: DB::Exception: Failed to wait RAFT initialization
2022.03.21 22:55:14.533422 [ 150503 ] {} <Information> Application: shutting down
2022.03.21 22:55:14.533513 [ 150504 ] {} <Information> BaseDaemon: Stop SignalListener thread
```

**Comment by wbxsGithub** [comment]
What's the root cause? Is there any problem with my configuration?

**Comment by alexey-milovidov** [comment]
Fixed by #40991.
